src/sdltiles.cpp

```diff
@@ -670,6 +676,372 @@ void clear_window_area( const catacurses::window &win_ )
+        geometry->rect( renderer, clipRect, SDL_Color() );
+    }
+
+    // color blocks overlay; drawn on top of tiles and on top of overlay strings (if any).
+    color_block_overlay_container color_blocks;
+    // Strings with colors do be drawn with map_font on top of tiles.
+    std::multimap<point, formatted_text> overlay_strings;
+    point s;
+    get_window_tile_counts( width, height, s.x, s.y );
+
+
+    op = point( dest.x * fontwidth, dest.y * fontheight );
+    // Rounding up to include incomplete tiles at the bottom/right edges
+    screentile_width = divide_round_up( width, tile_width );
+    screentile_height = divide_round_up( height, tile_height );
+
+    const int min_col = 0;
+    const int max_col = s.x;
+    const int min_row = 0;
+    const int max_row = s.y;
+    int height_3d = 0;
+    avatar &you = get_avatar();
+    const tripoint_abs_omt avatar_pos = you.global_omt_location();
+    const tripoint_abs_omt corner_NW = center_abs_omt - point( max_col / 2, max_row / 2 );
+    const tripoint_abs_omt corner_SE = corner_NW + point( max_col - 1, max_row - 1 );
+    const inclusive_cuboid<tripoint> overmap_area( corner_NW.raw(), corner_SE.raw() );
+    // Debug vision allows seeing everything
+    const bool has_debug_vision = you.has_trait( trait_id( "DEBUG_NIGHTVISION" ) );
+    // sight_points is hoisted for speed reasons.
+    const int sight_points = !has_debug_vision ?
+                             you.overmap_sight_range( g->light_level( you.posz() ) ) :
+                             100;
+    const bool showhordes = uistate.overmap_show_hordes;
+    const bool viewing_weather = uistate.overmap_debug_weather || uistate.overmap_visible_weather;
+    o = dest;
+
+    const auto global_omt_to_draw_position = [&corner_NW]( const tripoint_abs_omt & omp ) {
+        // z position is hardcoded to 0 because the things this will be used to draw should not be skipped
+        return tripoint( ( omp - corner_NW ).raw().xy(), 0 );
+    };
+
+    for( int row = min_row; row < max_row; row++ ) {
+        for( int col = min_col; col < max_col; col++ ) {
+            const tripoint pos( col + o.x, row + o.y, center_abs_omt.z() );
+            const tripoint_abs_omt omp = corner_NW + point( col, row );
+
+            const bool see = overmap_buffer.seen( omp );
+            const bool los = see && you.overmap_los( omp, sight_points );
+            const oter_id &cur_ter = overmap_buffer.ter( omp );
+            // the full string from the ter_id including _north etc.
+            std::string id;
+
+            if( viewing_weather ) {
+                const tripoint_abs_omt omp_sky( omp.xy(), OVERMAP_HEIGHT );
+                if( uistate.overmap_debug_weather ||
+                    you.overmap_los( omp_sky, sight_points * 2 ) ) {
+                    id = overmap_ui::get_weather_at_point( omp_sky ).c_str();
+                } else {
+                    id = "unknown_terrain";
```

Comment:
:warning: Clashes with fallback code in `cata_tiles.cpp`:
https://github.com/CleverRaven/Cataclysm-DDA/blob/b42ab9d79f788517a150ea32d0e9a9f777ca5a53/src/cata_tiles.cpp#L1873

Removed from Ultica in https://github.com/I-am-Erk/CDDA-Tilesets/pull/492 , still present at least in RetroDays

#47114

Reply:
`unexplored_`?